server/src/main/java/org/candlepin/dto/api/v1/ProductDTO.java

```diff
@@ -0,0 +1,963 @@
+     *
+     * @return
+     *  the dependent product IDs of the product, or null if the dependent product IDs have not yet
+     *  been defined
+     */
+    public Collection<String> getDependentProductIds() {
+        return this.dependentProductIds != null ? new SetView(this.dependentProductIds) : null;
+    }
+
+    /**
+     * Adds the ID of the specified product as a dependent product of this product. If the product
+     * is already a dependent product, it will not be added again.
+     *
+     * @param productId
+     *  The ID of the product to add as a dependent product
+     *
+     * @throws IllegalArgumentException
+     *  if productId is null
+     *
+     * @return
+     *  true if the dependent product was added successfully; false otherwise
+     */
+    public boolean addDependentProductId(String productId) {
+        if (productId == null) {
+            throw new IllegalArgumentException("productId is null");
+        }
+
+        if (this.dependentProductIds == null) {
+            this.dependentProductIds = new HashSet<String>();
+        }
+
+        return this.dependentProductIds.add(productId);
+    }
+
+    /**
+     * Removes the specified product as a dependent product of this product. If the product is not
+     * dependent on this product, this method does nothing.
+     *
+     * @param productId
+     *  The ID of the product to add as a dependent product
+     *
+     * @throws IllegalArgumentException
+     *  if productId is null
+     *
+     * @return
+     *  true if the dependent product was removed successfully; false otherwise
+     */
+    public boolean removeDependentProductId(String productId) {
+        return this.dependentProductIds != null ? this.dependentProductIds.remove(productId) : false;
+    }
+
+    /**
+     * Sets the dependent product IDs of the product represented by this DTO.
+     *
+     * @param dependentProductIds
+     *  A collection of dependent product IDs to attach to this DTO, or null to clear the
+     *  dependent products
+     *
+     * @return
+     *  a reference to this DTO
+     */
+    public ProductDTO setDependentProductIds(Collection<String> dependentProductIds) {
+        if (dependentProductIds != null) {
+            if (this.dependentProductIds == null) {
+                this.dependentProductIds = new HashSet<String>();
+            }
+            else {
+                this.dependentProductIds.clear();
+            }
+
+            for (String pid : dependentProductIds) {
+                this.addDependentProductId(pid);
+            }
+        }
+        else {
+            this.dependentProductIds = null;
+        }
+
+        return this;
+    }
+
+    /**
+     * Retrieves the link of the product represented by this DTO. If the product hyperlink has not
+     * yet been defined, this method returns null.
+     *
+     * @return
+     *  the link of the product, or null if the link have not yet been defined
+     */
+    public String getHref() {
```

Comment:
Couldn't this be defined dynamically?

Reply:
Technically yes, but I opted to decouple the DTOs from any logic in the model classes if I could. In this case, we *could* set the URL dynamically as Product does today, but if we were to change that behavior, we'd have to change it in two places.